### PR TITLE
pasta: Correct handling of unknown protocols

### PR DESCRIPTION
**Pull request by dgibson**
setupPasta() has logic to handle forwarding of TCP or UDP ports.  It has what looks like logic to give an error if trying to forward ports of any other protocol.  However, there's a straightforward error in this that it will in fact only give the error if you try to use a protocol called "default".  Other unknown protocols will fall through and result in a nonsensical pasta command line which will almost certainly cause a cryptic error later on.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by Luap99 — COMMENTED** [review]
Thanks, please make sure to a test for the expected error message, `test/system/505-networking-pasta.bats` would be a good place. Running `podman run -p 8080/sctp --network pasta ...` should be enough to trigger the bug reliably.

**Comment by dgibson** [comment]
> Thanks, please make sure to a test for the expected error message, `test/system/505-networking-pasta.bats` would be a good place. Running `podman run -p 8080/sctp --network pasta ...` should be enough to trigger the bug reliably.

Um.. yeah.  Writing such a test is great deal more complicated than this patch itself, especially when I don't know podman very well at all.   I'm unlikely to find time to do it.

**Comment by sbrivio-rh** [comment]
I'm not sure how I would attach a commit to this series, but here comes a patch adding the test suggested by @Luap99:

```
diff --git a/test/system/505-networking-pasta.bats b/test/system/505-networking-pasta.bats
index 032c9b0b8..989142e29 100644
--- a/test/system/505-networking-pasta.bats
+++ b/test/system/505-networking-pasta.bats
@@ -681,3 +681,11 @@ function teardown() {
     sleep 1
     ! ps -p $(cat "${pidfile}") && rm "${pidfile}"
 }
+
+### Options ####################################################################
+@test "podman networking with pasta(1) - Unsupported protocol in port forwarding" {
+    local port=$(random_free_port "" "" tcp)
+
+    run_podman 126 run --net=pasta -p "${port}:${port}/sctp" $IMAGE true
+    is "$output" "Error: can't forward protocol: sctp"
+}

```

we'll need to drop this the day we add [support for SCTP](https://bugs.passt.top/show_bug.cgi?id=3), but that would also need explicit support in the integration anyway.

**Comment by dgibson** [comment]
@sbrivio-rh, I've added your patch to my series and re-pushed.  Thanks for testcase.

File: test/system/505-networking-pasta.bats [hunk 1]
```diff
@@ -681,3 +681,11 @@ function teardown() {
     sleep 1
     ! ps -p $(cat "${pidfile}") && rm "${pidfile}"
 }
+
+### Options ####################################################################
+@test "podman networking with pasta(1) - Unsupported protocol in port forwarding" {
+    local port=$(random_free_port "" "" tcp)
+
+    run_podman 126 run --net=pasta -p "${port}:${port}/sctp" $IMAGE true
+    is "$output" "Error: can't forward protocol: sctp"
```
Comment (Luap99) [inline test/system/505-networking-pasta.bats:690]:
```suggestion
    is "$output" "Error: .*can't forward protocol: sctp"
```
The check is failing with podman-remote, see https://api.cirrus-ci.com/v1/artifact/task/6684547220242432/html/sys-remote-fedora-38-rootless-host-boltdb.log.html

Comment (dgibson) [reply test/system/505-networking-pasta.bats]: Ah, thanks.  I saw the failure but didn't know how to dig far enough into the logs to find a helpful error message.

**Review by Luap99 — APPROVED** [review]
LGTM, thanks

**Comment by baude** [comment]
@dgibson great to see you again ... thanks for the submission

/hold cancel

**Pull request merged**